homeassistant/components/knx/climate.py

```diff
@@ -166,6 +177,27 @@ def __init__(self, knx_module: KNXModule, config: ConfigType) -> None:
             self._attr_preset_modes = [
                 mode.name.lower() for mode in self._device.mode.operation_modes
             ]
+
+        if self._device.fan_speed is not None and self._device.fan_speed.initialized:
+            self._attr_supported_features |= ClimateEntityFeature.FAN_MODE
+            fan_max_step = config.get(ClimateSchema.CONF_FAN_MAX_STEP)
+
+            if fan_max_step is not None and fan_max_step == 3:
+                self._attr_fan_modes = [FAN_OFF, FAN_LOW, FAN_MEDIUM, FAN_HIGH]
+            elif fan_max_step is not None and fan_max_step == 2:
+                self._attr_fan_modes = [FAN_OFF, FAN_LOW, FAN_HIGH]
+            elif fan_max_step is not None and fan_max_step == 1:
+                self._attr_fan_modes = [FAN_OFF, FAN_ON]
+            elif fan_max_step is not None:
+                self._attr_fan_modes = [FAN_OFF] + [
+                    str(i) for i in range(1, fan_max_step + 1)
+                ]
+            else:
+                fan_percentages_modes = config[ClimateSchema.CONF_FAN_PERCENTAGES_MODES]
+                self._attr_fan_modes = [FAN_OFF] + [
+                    f"{value}%" for value in fan_percentages_modes
+                ]
```

Comment:
```suggestion
            if fan_max_step is None:
                fan_percentages_modes = config[ClimateSchema.CONF_FAN_PERCENTAGES_MODES]
                self._attr_fan_modes = [FAN_OFF] + [
                    f"{value}%" for value in fan_percentages_modes
                ]
            elif fan_max_step == 3:
                self._attr_fan_modes = [FAN_OFF, FAN_LOW, FAN_MEDIUM, FAN_HIGH]
            elif fan_max_step == 2:
                self._attr_fan_modes = [FAN_OFF, FAN_LOW, FAN_HIGH]
            elif fan_max_step == 1:
                self._attr_fan_modes = [FAN_OFF, FAN_ON]
            else:
                self._attr_fan_modes = [FAN_OFF] + [
                    str(i) for i in range(1, fan_max_step + 1)
                ]
```
switched order makes this slightly more readable